### PR TITLE
Add impl files to file check results

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This is port of https://github.com/fsharp/FSharp.Compiler.Service/pull/810 which adds `FSharpCheckFileResults.ImplementationFiles` that is needed by Fable compiler, see https://github.com/fsharp/FSharp.Compiler.Service/issues/796.

### CODE REVIEW

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test this please

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Release_ci_part2 please

**Comment by vasily-kirichenko** [comment]
It's ready.

File: src/fsharp/vs/service.fs [hunk 1]
```diff
@@ -2002,6 +2006,11 @@ type FSharpCheckFileResults(filename: string, errors: FSharpErrorInfo[], scopeOp
             RequireCompilationThread ctok
             scope.IsRelativeNameResolvable(pos, plid, item))
     
+    member info.ImplementationFiles =
+        scopeOptX
```
Comment (dsyme) [inline src/fsharp/vs/service.fs]: Could you add this please
```
        if not keepAssemblyContents then invalidOp "The 'keepAssemblyContents' flag must be set to tru on the FSharpChecker in order to access the checked contents of assemblies"
```

Comment (dsyme) [reply src/fsharp/vs/service.fs]: As for ``FSharpCheckProjectResults.AssemblyContents``

**Comment by dsyme** [comment]
Looks great - added one comment.  Please add a test too

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test this please

**Comment by vasily-kirichenko** [comment]
All done, it's ready as it's green.

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Release_ci_part1 please

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Windows_NT Release_ci_part1 Build please

File: src/fsharp/vs/service.fs [hunk 2]
```diff
@@ -2002,6 +2006,12 @@ type FSharpCheckFileResults(filename: string, errors: FSharpErrorInfo[], scopeOp
             RequireCompilationThread ctok
             scope.IsRelativeNameResolvable(pos, plid, item))
     
+    member info.ImplementationFiles =
+        if not keepAssemblyContents then invalidOp "The 'keepAssemblyContents' flag must be set to tru on the FSharpChecker in order to access the checked contents of assemblies"
```
Comment (auduchinok) [inline src/fsharp/vs/service.fs]: A typo in `true`.

Comment (vasily-kirichenko) [reply src/fsharp/vs/service.fs]: Thanks, done.

**Comment by vasily-kirichenko** [comment]
Any hope it'll be merged?

**Comment by dsyme** [comment]
(We will need to make a new FCS release)

**Comment by vasily-kirichenko** [comment]
@dsyme I made an equivalent PR there, but I think all common files should be synchronized?

**Comment by dsyme** [comment]
> @dsyme I made an equivalent PR there, but I think all common files should be synchronized?

That's not needed - we can just integrate across.  Also FCS nugets can be built from this repo (I'll double check that)

**Comment by auduchinok** [comment]
@dsyme We build it from this repo, it works fine.

**Comment by vasily-kirichenko** [comment]
@dsyme Actually we've been waiting for FCS maintainers to publish a new package (we - Fable).

**Comment by dsyme** [comment]
@vasily-kirichenko Published as FCS 15.0.1

**Pull request merged**